tenable/io/v3/vm/agents/api.py

```diff
@@ -0,0 +1,138 @@
+'''
+Agents
+======
+
+The following methods allow for interaction into the Tenable.io
+:devportal:`agents <agents>` API endpoints.
+
+Methods available on ``tio.v3.vm.agents``:
+
+.. rst-class:: hide-signature
+.. autoclass:: AgentsAPI
+    :members:
+'''
+from typing import Dict, Union
+from uuid import UUID
+
+from tenable.io.v3.base.endpoints.explore import ExploreBaseEndpoint
+from tenable.io.v3.vm.agents.schema import AgentsSchema
+
+
+class AgentsAPI(ExploreBaseEndpoint):
+    _path: str = 'api/v3/agents'
+    _conv_json: bool = True
+
+    def details(self, agent_id: UUID) -> Dict:
+        '''
+        Retrieves the details of an agent.
+
+        :devportal:`agents: get <agents-get>`
+
+        Args:
+            agent_id (UUID): The unique identifier of the agent.
+
+        Returns:
+            :obj:`dict`:
+                The agent dictionary record.
+
+        Examples:
+            >>> agent = tio.v3.vm.agents.details(
+            ...     '00000000-0000-0000-0000-000000000000'
+            ... )
+            >>> pprint(agent)
+        '''
+        return self._get(f'{agent_id}')
+
+    def list_agents_from_group(self, agent_group_id: UUID) -> Dict:
+        '''
+        Retrive the list of agents for the specified agent group.
+
+        :devportal:`agent: get <agent-group-list-agents>`
+
+        Args:
+            agent_group_id (UUID): The unique identifier of the agent.
+
+        Returns:
+            :obj:`dict`:
+                The agent groups dictionary of all the agents.
+
+        Examples:
+            >>> agent = tio.v3.vm.agents.list_agents_from_groups(
+            ...     '00000000-0000-0000-0000-000000000000'
+            ... )
+            >>> print(agent_from_group)
+        '''
+        self._path: str = self._path.replace('agents', 'agent-groups')
```

Comment:
@milanaghera-tenable I think we should avoid overwriting self._path as it might mean that once this method is called, the self._path will be updated. And this might lead to an unexpected error. We could use self._api... directly in this case.